ml-agents/mlagents/trainers/torch/distributions.py

```diff
@@ -114,7 +124,10 @@ def all_log_prob(self):
         return torch.log(self.probs)
 
     def entropy(self):
-        return -torch.sum(self.probs * torch.log(self.probs), dim=-1)
+        return -torch.sum(self.probs * torch.log(self.probs), dim=-1).unsqueeze(-1)
+
+    def exported_model_output(self):
+        return self.all_log_prob()
```

Comment:
I think that only the legacy output should be the all_log_prob. The new action output should be the sampled actions. This way we can directly use Barracuda to sample the actions instead of relying on our own implementation of multicategorical.

Reply:
Would we need to change the C# side to do that?

Reply:
I think this is out of scope for this PR